packages/core/src/hooks/useSigner.tsx

```diff
@@ -0,0 +1,15 @@
+import { useEthers } from './useEthers'
+import { useState, useEffect } from 'react'
+import { JsonRpcProvider, JsonRpcSigner } from '@ethersproject/providers'
+
+export function useSigner(): JsonRpcSigner | undefined {
```

Comment:
If you want to release it to the public, you need:
- a changeset
- Add JSDOC
    - With some human readable comments that will end up in docs - check that they are properly auto-generated in the API reference section
    - with a `@public` annotation
- Export it in https://github.com/TrueFiEng/useDApp/blob/master/packages/core/src/index.ts
    - We don't export `*` wildcards in order to consciously control public API contract